### ISSUE TITLE
Is it possible to create slides based on response data from API?

### ISSUE THREAD

**Issue by wiadev** [alvarotrigo/angular-fullpage#27]
Hi
I'm trying to build a fullpage.js slide with the data from the api.

Here is my code snippet.

```
<div *ngIf="items">
    <div fullpage id="fullpage" [options]="config" (ref)="getRef($event)" #fullpageRef>
      <div class="section" *ngFor="let item of items; let idx = index;">
        <h1>{{idx}}</h1>
        <h2>{{item.id}}</h2>
      </div>
    </div>
  </div>
```

But got the following error messages.

![screenshot at jan 21 00-32-31](https://user-images.githubusercontent.com/18234060/51461960-4a9f0e00-1d14-11e9-9939-7892595a2285.png)

Any thoughts on this?
Thanks

**Comment by alvarotrigo** [comment]
Can you try again by replacing the extensions file for the normal one and see if you still have the error?

In `angular-fullpage/projects/angular-fullpage/src/lib/fullpage.directive.ts` replace:

> import fullpage from 'fullpage.js/dist/fullpage.extensions.min';

For 

> import fullpage from 'fullpage.js/dist/fullpage.js';

**Comment by wiadev** [comment]
Thanks for your response.

![screenshot at jan 21 02-53-13](https://user-images.githubusercontent.com/18234060/51470116-d0788480-1d27-11e9-8f17-6a482786d3d5.png)

Here is the @fullpage/angular-fullpage npm package.

Could you please explain to me which file to edit? the file you mentioned is not available.

**Comment by alvarotrigo** [comment]
Oh, of course, not in npm :)
I was talking about [this file](https://github.com/alvarotrigo/angular-fullpage/blob/master/projects/angular-fullpage/src/lib/fullpage.directive.ts) you can get from this github repo in order to build the dist files.

Would you be able to reproduce or recreate your issue in this [Stackblitz](https://stackblitz.com/edit/angular-nqsqwk)?

**Comment by vh13294** [comment]
Please use read this to dynamically add sections.
You cannot use ngFor because fullpage.js will start to build the dom as soon as angular compent is created.

Doc: https://github.com/alvarotrigo/angular-fullpage#dynamic-changes 
Example: https://github.com/alvarotrigo/angular-fullpage/blob/master/src/app/examples/dynamic-changes/dynamic-changes.component.ts#L46

**Comment by vh13294** [comment]
@wiadev if you really need this.
I can easily implement a new feature to support dynamic *ngFor! by this weeken!

**Comment by wiadev** [comment]
Great! @vh13294 

Yes, I believe this is the common issue for angular developers since they will create context from RESTful API.

Is it possible to implememt in a few days?
If not, I have to go ahead with dynamic approach.

Another question for adding dynamic content.

```
this.renderer.setProperty(section, 'innerHTML', '<h3>New Section</h3>');
```
Can I add custom component instead of HTML default tags?

Thanks

**Comment by vh13294** [comment]
I am not sure that's going to work. But you can give it a try like pass a reference of fullpage.js to your child component. Then on event afterViewInit you can rebuild ```fullpage_api.build();```

**Comment by wiadev** [comment]
When do you think it will be released in public?

**Comment by alvarotrigo** [comment]
@wiadev I'll take a look at it tomorrow and hopefully I can merge it then.

**Comment by alvarotrigo** [comment]
Now fixed in the latest release 0.0.10!
Thanks @vh13294 for it!
